### DISCUSSION TITLE
Addressing and Correcting Code Issues in CasADI's Python Examples at the Home page

### DISCUSSION THREAD

**Discussion by jasonr18** [casadi/casadi#3210] [General]
Hello,

I'm a newcomer to CasADI, having previously worked with Julia and JuMP. I've found CasADI's symbolic framework quite intuitive. However, I noticed some issues when attempting to follow the Python examples provided on the homepage. For the benefit of other new users, I believe these issues should be addressed.

In the "Nonlinear and quadratic programming" example, the function `f = x2+100*z2` appears to be erroneous. Based on my understanding, it should likely be written as `f = x**2+100*z**2`.

Moreover, in the "Composition of the above" example, there are a few discrepancies concerning the formulations, indentation, and plots. I have taken the liberty to revise the Python code accordingly. It goes as follows:

``` python
from casadi import *
from matplotlib.pyplot import plot, show
x = MX.sym('x',2); # Two states
p = MX.sym('p');   # Free parameter


# Expression for ODE right-hand side
z = 1-x[1]**2
rhs = vertcat(z*x[0]-x[1]+2*tanh(p),x[0])


# ODE declaration with free parameter
ode = {'x':x,'p':p,'ode':rhs}


# Construct a Function that integrates over 1s
F = integrator('F','cvodes',ode,{'tf':1})


# Control vector
u = MX.sym('u',4,1)


x = [0,1]  # Initial state
for k in range(4):
# Integrate 1s forward in time:
# call integrator symbolically
    res = F(x0=x,p=u[k])
    x = res["xf"]


# NLP declaration
nlp = {'x':u,'f':dot(u,u),'g':x}


# Solve using IPOPT
solver = nlpsol('solver','ipopt',nlp)
res = solver(x0=0.2,lbg=0,ubg=0)


plot(res["x"])
show()
```

**Comment by jaeandersson** [comment]
Thanks! I think the power issue has been pointed out in another issue recently - it's some sort of rendering issue. The same is probably true for the second issue. Ping @jgillis

**Comment by jgillis** [comment]
Thank you for notifying us, web page should be updated now.
